src/components/search/form/clauses/FileSize.js

```diff
@@ -0,0 +1,124 @@
+import buildID from "components/utils/DebugIDUtil";
+import { minValue } from "components/utils/validations";
+import { useTranslation } from "i18n";
+import ids from "../ids";
+
+const SIZE_TYPE = "size";
+const SIZE_ARGS_DEFAULT = {
+    from: { value: "", unit: "KiB" },
+    to: { value: "", unit: "KiB" },
+};
+
+// remove empty values, merge size and unit to one string
+const formatFileSizeValues = (clause) => {
+    // create copy of the clause, don't modify the original otherwise it
+    // mutates the form
+    let filteredValues = JSON.parse(JSON.stringify(clause));
+    let args = filteredValues.args;
+
+    if (!args.from.value && !args.to.value) {
+        return null;
+    }
+
+    if (args.from.value) {
+        filteredValues.args.from = `${args.from.value}${args.from.unit}`;
+    } else {
+        delete filteredValues.args.from;
+    }
+    if (args.to.value) {
+        filteredValues.args.to = `${args.to.value}${args.to.unit}`;
+    } else {
+        delete filteredValues.args.to;
+    }
+
+    return filteredValues;
+};
+
+function FileSize(props) {
+    const {
+        parentId,
+        field: { name },
+    } = props;
+
+    const { t } = useTranslation("search");
+
+    return (
+        <Grid container spacing={1}>
+            <Grid item>
+                <FastField
+                    name={`${name}.from.value`}
+                    helperText={t("fileSizeGreater")}
+                    id={buildID(parentId, ids.FILE_SIZE_GREATER_VAL)}
+                    validate={(value) => minValue(value, t)}
+                    fullWidth={false}
+                    component={FormNumberField}
+                />
+            </Grid>
+            <Grid item>
+                <FastField
+                    name={`${name}.from.unit`}
+                    id={buildID(parentId, ids.FILE_SIZE_GREATER_UNIT)}
+                    label=" "
+                    component={SizeUnit}
+                />
+            </Grid>
+            <Grid item>
+                <FastField
+                    name={`${name}.to.value`}
+                    helperText={t("fileSizeLessThan")}
+                    id={buildID(parentId, ids.FILE_SIZE_LESS_THAN_VAL)}
+                    validate={(value) => minValue(value, t)}
+                    fullWidth={false}
+                    component={FormNumberField}
+                />
+            </Grid>
+            <Grid item>
+                <FastField
+                    name={`${name}.to.unit`}
+                    id={buildID(parentId, ids.FILE_SIZE_LESS_THAN_UNIT)}
+                    label=" "
+                    component={SizeUnit}
+                />
+            </Grid>
+        </Grid>
+    );
+}
+
+const sizesList = ["KiB", "MiB", "GiB", "TiB"];
+
+function SizeUnit(props) {
+    const {
+        field,
+        form: { setFieldValue, ...form },
+        ...rest
+    } = props;
+    return (
+        <FormSelectField form={form} field={field} fullWidth={false} {...rest}>
```

Comment:
I don't see why `setFieldValue` should be removed from the `FormSelectField` props.

Alternatively, rather than using a `FormSelectField`, you could also use a `FormTextField` here with the `select` prop set (I'd also suggest setting `variant="outlined"` too).

For example:
https://github.com/cyverse-de/sonora/blob/45d876316a3bf625f3a832cb61f7cea5f8e73b23/src/components/apps/launch/params/Selection.js#L12-L23

Reply:
While this works 
![image](https://user-images.githubusercontent.com/1250790/146046250-8ee10037-8cd4-4a36-acf1-2c5c109f502f.png)

It will nice to have something like what google drive does for Advanced search / filter:

![image](https://user-images.githubusercontent.com/1250790/146046408-68f57497-500d-4a07-bef8-bedc1553739c.png)


May be to the left of `Advanced Data Search` button 🤔